controllers/feeds/detailFeed.js

```diff
@@ -22,7 +24,11 @@ module.exports = async (req, res) => {
     return ErrorResponse.e404(res, 'This post has expired and has been deleted automatically');
   }
 
+  const myAnonymousUser = await UsersFunction.findAnonymousUserId(User, req.userId, {raw: true});
+
   const newItem = {...feedItem};
+  newItem.is_self =
+    newItem.actor.id === req.userId || newItem.actor.id === myAnonymousUser?.user_id;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `is_self` field is being added to the `newItem` object, which is a copy of `feedItem`. However, it's not clear if this change will persist beyond the scope of this function. If `newItem` is used later in the code for further operations or responses, then this change is fine. But if `feedItem` is used instead, then the `is_self` field won't be present. Please ensure that the correct object (`newItem`) is used throughout the rest of the code after this point.

Also, consider handling the case where `myAnonymousUser` is `null` or `undefined`. Currently, if `UsersFunction.findAnonymousUserId()` doesn't find an anonymous user, it might return `null` or `undefined`, and trying to access `user_id` on it would throw an error. You can add a check before accessing `user_id` to prevent this.

```diff
- const myAnonymousUser = await UsersFunction.findAnonymousUserId(User, req.userId, {raw: true});
+ const myAnonymousUser = await UsersFunction.findAnonymousUserId(User, req.userId, {raw: true}) || {};
```

This way, if `findAnonymousUserId()` returns `null` or `undefined`, `myAnonymousUser` will default to an empty object, and accessing `user_id` on it will simply return `undefined` without throwing an error.


<!-- This is an auto-generated comment by OSS CodeRabbit -->